lightningd/payalgo.c

```diff
@@ -215,11 +238,15 @@ static bool json_pay_try(struct pay *pay)
 		return false;
 	}
 
-	/* Clear previous sendpay. */
+	/* Clear previous try memory. */
 	pay->try_parent = tal_free(pay->try_parent);
 	pay->try_parent = tal(pay, char);
 
-	++pay->tries;
+	/* Generate random seed */
+	seed = tal_arr(pay->try_parent, u8, ISAAC64_SEED_SZ_MAX);
```

Comment:
2048 bytes of randonmess is extreme!

8 bytes of non-cryptographic randomness is more than sufficient; we should just add a psuedorand64() which wraps isaac64_next_uint64() and use that here I think.

Reply:
Okay, will do.